Refactor to objects.

davidmorgan opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/built_value.dart/30)
<!-- Reviewable:end -->

manuelF commented:
Review status: 0 of 10 files reviewed at latest revision, 3 unresolved discussions.

---

*[built_value_generator/lib/src/source_class.dart, line 111 \[r1\]](https://reviewable.io:443/reviews/google/built_value.dart/30#-KPIT3J3ZUUIHcNMXBR_:-KPIT3J3ZUUIHcNMXBRa:b14hx77) ([raw file](https://github.com/google/built_value.dart/blob/05f6a646b97790c179570ae8e05eaa6e97610114/built_value_generator/lib/src/source_class.dart#L111)):*
> ```Dart
>     final expectedConstructor = '$name._()';
>     if (valueClassConstructors.length != 1 ||
>         //constructors.single.isSynthetic ||
> ```

is it supossed to be a comment?

---

*[built_value_generator/lib/src/source_class.dart, line 140 \[r1\]](https://reviewable.io:443/reviews/google/built_value.dart/30#-KPITB93-RRkvLaazkfb:-KPITB93-RRkvLaazkfc:bb73xfv) ([raw file](https://github.com/google/built_value.dart/blob/05f6a646b97790c179570ae8e05eaa6e97610114/built_value_generator/lib/src/source_class.dart#L140)):*
> ```Dart
>     final expectedConstructor = '${name}Builder._()';
>     if (builderClassConstructors.length != 1 ||
>         //constructors.single.isSynthetic ||
> ```

Same

---

*[built_value_generator/lib/src/source_field.dart, line 85 \[r1\]](https://reviewable.io:443/reviews/google/built_value.dart/30#-KPITd58PXbtkTBUQ5qh:-KPITd58PXbtkTBUQ5qi:bq8afnp) ([raw file](https://github.com/google/built_value.dart/blob/05f6a646b97790c179570ae8e05eaa6e97610114/built_value_generator/lib/src/source_field.dart#L85)):*
> ```Dart
>   }
> 
>   /*static BuiltList<SourceField> fromBuilderClassElement(
> ```

comment?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/built_value.dart/30)*
<!-- Sent from Reviewable.io -->

davidmorgan commented:
Review status: 0 of 11 files reviewed at latest revision, 3 unresolved discussions.

---

*[built_value_generator/lib/src/source_class.dart, line 111 \[r1\]](https://reviewable.io:443/reviews/google/built_value.dart/30#-KPIT3J3ZUUIHcNMXBR_:-KQ4svL_69Sl5A8IeSGB:bh0km72) ([raw file](https://github.com/google/built_value.dart/blob/05f6a646b97790c179570ae8e05eaa6e97610114/built_value_generator/lib/src/source_class.dart#L111)):*
<details><summary><i>Previously, manuelF (Manuel Ferreria) wrote…</i></summary>
> is it supossed to be a comment?
</details>
Ah, missed this case. The problem is if you have no constructor/factory then a synthetic constructor appears.

Fixed by removing this comment, and when we constructor valueClassConstructors, drop synthetic constructors. Also added test coverage for this case. (As was, it was an NPE due to trying to read source for the synthetic constructor).

---

*[built_value_generator/lib/src/source_class.dart, line 140 \[r1\]](https://reviewable.io:443/reviews/google/built_value.dart/30#-KPITB93-RRkvLaazkfb:-KQ4u2P0W8u0BhbyCDTK:b96nu81) ([raw file](https://github.com/google/built_value.dart/blob/05f6a646b97790c179570ae8e05eaa6e97610114/built_value_generator/lib/src/source_class.dart#L140)):*
<details><summary><i>Previously, manuelF (Manuel Ferreria) wrote…</i></summary>
> Same
</details>
Thanks, same fix, added test case.

---

*[built_value_generator/lib/src/source_field.dart, line 85 \[r1\]](https://reviewable.io:443/reviews/google/built_value.dart/30#-KPITd58PXbtkTBUQ5qh:-KQ4uEBNcxUMPw7VgNYt:bss4a2b) ([raw file](https://github.com/google/built_value.dart/blob/05f6a646b97790c179570ae8e05eaa6e97610114/built_value_generator/lib/src/source_field.dart#L85)):*
<details><summary><i>Previously, manuelF (Manuel Ferreria) wrote…</i></summary>
> comment?
</details>
Removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/built_value.dart/30)*
<!-- Sent from Reviewable.io -->

manuelF commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 11 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/built_value.dart/30#-:-KQ59HMzKE9rQ6z5UBI-:bnfp4nl)*
<!-- Sent from Reviewable.io -->
